src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -4074,9 +4074,17 @@ GenTree* Lowering::LowerHWIntrinsicCndSel(GenTreeHWIntrinsic* cndSelNode)
         // `trueValue`
         GenTreeHWIntrinsic* nestedCndSel = op2->AsHWIntrinsic();
         GenTree*            nestedOp1    = nestedCndSel->Op(1);
+        GenTree*            nestedOp2    = nestedCndSel->Op(2);
         assert(varTypeIsMask(nestedOp1));
+        assert(nestedOp2->OperIsHWIntrinsic());
 
-        if (nestedOp1->IsMaskAllBitsSet())
+        NamedIntrinsic nestedOp2Id = nestedOp2->AsHWIntrinsic()->GetHWIntrinsicId();
+
+        // If the nested op uses Pg/Z, then inactive lanes will result in zeros, so can only transform if
+        // op3 is all zeros.
+
+        if (nestedOp1->IsMaskAllBitsSet() &&
+            (!HWIntrinsicInfo::IsZeroingMaskedOperation(nestedOp2Id) || op3->IsVectorZero()))
```

Comment:
Are there any instructions that only allow `Pg/M`?

Basically we have:
* Pg/Z only
* Pg/M -or- Pg/Z

So I'm wanting to discern if we also have:
* Pg/M only

Reply:
Yes, quite a few.
eg:
https://docsmirror.github.io/A64/2023-09/add_z_p_zz.html
`ADD <Zdn>.<T>, <Pg>/M, <Zdn>.<T>, <Zm>.<T>`

Reply:
We likely need something that indicates that and avoids containment of zero for that case then, correct?

Reply:
If it's `<Pg>/M` only then the optimisation will continue to work the same as it does in HEAD (because `IsZeroingMaskedOperation()` will be false)